diff --git a/device/downstream/linux-samsung-gtelwifi/APKBUILD b/device/downstream/linux-samsung-gtelwifi/APKBUILD
index 2c728bc07..d27cc270c 100644
--- a/device/downstream/linux-samsung-gtelwifi/APKBUILD
+++ b/device/downstream/linux-samsung-gtelwifi/APKBUILD
@@ -28,7 +28,7 @@ source="
 	linux3.0-8933-1-replace-Sun-Solaris-style-flag-on-section.patch
 	linux3.4-fix-piggy.gzip.S.patch
 	fix-recordmcount.patch
-	fix-dtb_qcom,msm-id.patch
+	fix-dtb_qcom-msm-id.patch
 	sprdfb-check-for-buffering.patch
 	sprdfb-fix-swapped-colors.patch
 "
@@ -64,7 +64,7 @@ ad0182a483791fc88e058838bc331b2f04a75ba291e763767babdb815efadfc3b4fda97e69e2e3f0
 66ab793bb1fa50a07f10e2cdb0f49ea05a787c6b1fe5358e69a76e43295ac5be3b68cdf6a882e868bfe6969cac981c5545587cd57eee4b5cec26078ae0ecaa1c  linux3.0-8933-1-replace-Sun-Solaris-style-flag-on-section.patch
 080745ca4ecaa804ee1df26ea81975ca93b74372e0c08322a6c481e97807495ae6ceb6cc77653afa24f9f2df45da8ebf8e5de6b06037fbb08e0279877ce42188  linux3.4-fix-piggy.gzip.S.patch
 32911549f0c89972d06af838954978ee90fb2eabacc1393c0bffa12deb9d80ddec5322f859df772d90afea733965bd669e85d5e90657d4f69995ae9fc25dd4f3  fix-recordmcount.patch
-e8137de3cc4d49d5447cad6c299f0f88c2d2e60e012168bc857f4898ebc1da5c88c679927227c8675211af079d5a56b5b1baef4a3dbc86cb4e921c14d3f93357  fix-dtb_qcom,msm-id.patch
+e8137de3cc4d49d5447cad6c299f0f88c2d2e60e012168bc857f4898ebc1da5c88c679927227c8675211af079d5a56b5b1baef4a3dbc86cb4e921c14d3f93357  fix-dtb_qcom-msm-id.patch
 c985b4a320904c0c66fb976a4f66dd2671f21d7b86ea98c019684efc41b5801f4b9c23012f4aa37bc5ce3efd70a30523c9c78fcd6e5fbbcff172ff19d11a6579  sprdfb-check-for-buffering.patch
 6836001b24508d413a059f19b0c6f6f547037f68886c22159ad51be91d1a089e3d1d9ed79c3c62724648df7f810cd81d96a83f4a34368b373524b649e5539656  sprdfb-fix-swapped-colors.patch
 "
diff --git a/device/downstream/linux-samsung-gtelwifi/fix-dtb_qcom,msm-id.patch b/device/downstream/linux-samsung-gtelwifi/fix-dtb_qcom-msm-id.patch
similarity index 100%
rename from device/downstream/linux-samsung-gtelwifi/fix-dtb_qcom,msm-id.patch
rename to device/downstream/linux-samsung-gtelwifi/fix-dtb_qcom-msm-id.patch
