diff --git a/deviceinfo_schema.toml b/deviceinfo_schema.toml
index 0f90e40e0..12db306e1 100644
--- a/deviceinfo_schema.toml
+++ b/deviceinfo_schema.toml
@@ -266,6 +266,7 @@ description = "Additional custom arguments to pass to mkbootimg. Caution: adding
 
 [variable.flash.header_version]
 description = "Header version of boot image as defined by Android documentation."
+datatype = "integer"
 integer_interval = "[0, 4]"
 
 [variable.flash.generate_bootimg]
